scripts/vsc-venv-wrapper/vsc-venv.sh

```diff
@@ -0,0 +1,252 @@
+  echo ""
+  echo "  -d, --deactivate        Deactivate the virtual environment."
+  echo ""
+  echo "  -h, --help              Show this help message and exit."
+  echo "  -v, --version           Show the version information and exit."
+  echo ""
+  echo "Example Usage:"
+  echo "  $ source $SCRIPT_NAME --activate --requirements requirements.txt --modules modules.txt"
+  echo "  $ python my_script.py"
+  echo "  $ source $SCRIPT_NAME --deactivate"
+  return 1
+}
+
+
+echo_info() { echo -e "\e[32m$SCRIPT_NAME: [INFO] $1\e[0m"; }
+echo_warning() { echo -e "\e[33m$SCRIPT_NAME: [WARNING] $1\e[0m"; }
+echo_error() { echo -e "\e[31m$SCRIPT_NAME: [ERROR] $1\e[0m"; }
+
+load_modules() {
+  local modules_file lines
+  modules_file="$1"
+
+  if ! mapfile -t lines < "$modules_file"; then # Read the file line by line into an array
+    echo_error "Could not read modules script '$modules_file'"
+    return 1
+  fi
+
+  for line in "${lines[@]}"; do # Loop over each line in the array
+
+    # Skip empty lines
+    if [[ -z "$line" ]]; then
+      continue
+    fi
+
+    if module load $line; then
+      echo_info "  ✔ Module '$line' loaded successfully"
+    else
+      echo_error "Could not load module '$line'"
+      return 1
+    fi
+
+  done
+
+}
+
+# Description:
+#   This function checks if the currently loaded cluster's operating system and architecture are the same
+#   as the current host's OS and architecture.
+#   For example, switching to the shinx cluster on a gallade host will create a mismatch
+#   (RHEL8-zen2 on gligar and RHEL9-zen4 on shinx at the time of writing)
+#
+# Return:
+#   0 (true) - If the loaded cluster's OS and architecture match the host's OS and architecture.
+#   1 (false) - If there is a mismatch between the loaded and current host's OS or architecture.
+is_loaded_cluster_compatible_with_host() {
+  local loaded_cluster_os loaded_cluster_architecture
+  local current_host_os current_host_architecture
+
+  loaded_cluster_os="$VSC_OS_LOCAL"
+  loaded_cluster_architecture="$VSC_ARCH_LOCAL"
+
+  current_host_os=$(ml show env/vsc/"$VSC_DEFAULT_CLUSTER_MODULE" | grep VSC_OS_LOCAL | cut -d'"' -f4)
+  current_host_architecture=$(ml show env/vsc/"$VSC_DEFAULT_CLUSTER_MODULE" | grep VSC_ARCH_LOCAL | cut -d'"' -f4)
+
+  if [ "$loaded_cluster_os-$loaded_cluster_architecture" = "$current_host_os-$current_host_architecture" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+
+# ============================ Main functions ============================
+
+activate() {
+  local requirements_file modules_file
+  local venv_location n_loaded_modules python_version
+
+  requirements_file="$1"
+  modules_file="$2"
+
+  venv_location=$(realpath -m "venvs/venv-${VSC_OS_LOCAL}-${VSC_ARCH_LOCAL}") # full path of venv
+
+  # === Check if requirements file exists === #
+  if [ ! -f "$requirements_file" ]; then
+    echo_error "Requirements file '$requirements_file' not found"
```

Comment:
mention `--requirements` here:

```suggestion
    echo_error "Requirements file '$requirements_file' not found, use --requirements or -r to specify correct path"